st2common/st2common/util/pack_management.py

```diff
@@ -249,6 +249,12 @@ def clone_repo(temp_dir, repo_url, verify_ssl=True, ref='master'):
         repo.git.checkout('-b', short_branch, branch)
         branch = repo.head.reference
     else:
+        # Checkout out the latest tag if tag exists when branch or version is not specified.
+        if not ref and repo.tags:
+            branch = repo.git.describe('--abbrev=0', '--tags')
```

Comment:
This is not a valid source for truth for getting a "latest" tag.

It should rely first on https://index.stackstorm.org/v1/index.json, - the latest version in package index that associated with the pack.

Reply:
Here is a pointer for more native fix: 
https://github.com/StackStorm/st2/blob/c70b2dbfd311f34ea98a25f9ac0b5a4e15141f12/st2common/st2common/util/pack_management.py#L108
Getting the correct version there would be better.